warcprox/mitmproxy.py

```diff
@@ -134,6 +135,10 @@ def close(self):
     def __len__(self):
         return self.len
 
+    @property
+    def duration(self):
+        return round(time.time() - self.start_time, 3)
+
```

Comment:
I wouldn't put this here. It's weird partly because it never stops counting once it stops. Instead I would say `start = time.time()` at the beginning of `proxy_request()` and replace this line

```
485                         prox_rec_res.recorder.duration > self._max_request_duration):
```
with this
```
485                         time.time() - start > self._max_request_duration):
```